### ISSUE TITLE
Multicluster support for Helm via Flux packages in current architecture

### ISSUE THREAD

**Issue by castelblanque** [vmware-tanzu/kubeapps#5342]
**Summary**
Add multicluster support for managing Flux packages extending current architecture.

**Background and rationale**
After the [current multicluster support analysis](https://github.com/vmware-tanzu/kubeapps/blob/main/site/content/docs/latest/background/multicluster.md), it is clear that Carvel and Flux don't really have multicluster support.
Only Helm direct plugin supports almost all operations.

Work needs to be done in order to achieve some minimal level of multicluster usage with Flux plugin. Mainly work on the source code, as existing API should not need any changes at first sight.

**Description**
Make the changes in code required to implement the following features in a remote cluster without Kubeapps installed:

- ~~List available packages from global repository~~
- ~~List available packages from namespaced repository~~
- List installed packages in namespace
- Package management (install, delete, etc.)
- Get installed package details

Listing packages from a remote cluster is not possible as Flux plugin makes use of Redis as cache for the index.

**Acceptance criteria**

- All listed features in the **Description** section have been implemented

**Additional context**
There is some discussion about the operation model for multicluster in the [Multicluster support evolution for Kubeapps document](https://docs.google.com/document/d/1feK3VLJyf-5V7EnLgf0DbaeE47Zf4poHWYQ1_7GHJVI/edit?pli=1#bookmark=id.qztg0nookemr). However, for this issue the current architecture shouldn't change.

**Comment by dlaloue-vmware** [comment]
what is the architecture?
the issue title says "in current architecture", so i understand it as flux/carvel being installed on the manager cluster, not the target clusters.

if flux/carvel are on the manager cluster, then listing global packages is possible - though it is crossed out.
if flux/carvel are on the target cluster, then this is not the "current architecture".
can you clarify?

**Comment by castelblanque** [comment]
> so i understand it as flux/carvel being installed on the manager cluster, not the target clusters

Exactly, in the current architecture there is no Flux/Carvel in the workload clusters.
However, we decided that for the new architecture there will be Flux/Carvel, so maybe we want to hold or withdraw this issue to avoid doing work in vain?

The goal of this issue is to use the mechanism in both Carvel and Flux to provide them with a kubeconfig that will connect to the workload cluster when reconciling. For that, we should have a token available somewhere in the management cluster. Maybe extending the RBAC of `clusters[1].serviceToken`? That is up to the implementation I would say.

**Comment by castelblanque** [comment]
Closing issue because it is a complete different approach to the new multicluster design.
